### ISSUE TITLE
JIT + fprintf() = :(

### ISSUE THREAD

**Issue by Akuli** [Akuli/jou#147]
The tests on #146 are failing, because the LLVM JIT picks up `fprintf()` from `C:\Windows\system32\msvcrt.dll`, but `fopen()`, `fclose()`, `fputs()` and most other IO functions from LLVM's `ucrtbase.dll` (e.g. `C:\Program Files\LLVM\bin\ucrtbase.dll`). Apparently telling the JIT to just pick up a function named `fprintf` from somewhere is a Bad Idea: https://lists.llvm.org/pipermail/llvm-dev/2019-January/128809.html

I think I will next work on compiling Jou code into executable files, and hope that this bug is JIT specific like I think it is. I don't really need a JIT, it was just the second way to get the code running I figured out. (The first way was quite hacky.)

**Comment by Akuli** [comment]
Turns out LLVM by itself doesn't have a linker, and the only way to link something is to run the linker in a subprocess. https://stackoverflow.com/a/11692488

On Linux, I was going to just invoke `ld`, but after compiling a hello world into an object file `a.o`, a simple `ld a.o` didn't work at all. Then I looked at how clang runs `ld`, with `clang --verbose a.o`, and the command is about 1 kilometre long. I guess I'll just invoke `clang` and let it generate the `ld` command...

I haven't yet figured out how to find and run a linker on Windows.
